### ISSUE TITLE
Connection to node -1 (kafka/172.22.0.2:29092) could not be established. Broker may not be available.

### ISSUE THREAD

**Issue by ArpadGBondor** [obsidiandynamics/kafdrop#504]
Hi, I'm learning Kafka, and I tried to follow this tutorial: https://www.youtube.com/watch?v=JJEKPqSlXvk

When I try to start this project with `docker-composer up`, this is the error message I keep getting:

```
2023-04-11 08:51:09 2023-04-11 07:51:09.169  WARN 1 [| kafdrop-admin] o.a.k.c.NetworkClient                    : [AdminClient clientId=kafdrop-admin] Connection to node -1 (kafka/172.22.0.2:29092) could not be established. Broker may not be available.
```

I'm a bit clueless about how this project is supposed to work. Do you have any idea what went wrong?

**Comment by mehdihasan** [comment]
Hi @ArpadGBondor, 

As mentioned in the project intro, Kafdrop is a web UI for viewing Kafka topics and browsing consumer groups. The tool displays information such as brokers, topics, partitions, consumers, and lets you view messages.

**To run this project effectively, you need to make sure Kafka broker is up and running.** 

In the [docker compose](https://github.com/obsidiandynamics/kafdrop/blob/master/docker-compose/kafka-kafdrop/docker-compose.yaml) file, you run both a Kafka broker and Kafdrop. In some scenarios, Kafka broker may fail to run for some reason thus the Kafka container stopped. In such cases, you can get the above error. 

**_Basically the the log was saying that Kafdrop was expecting Kafka broker to available at port 29092 - which is actually not available._** 

You may need to follow the Kafka logs to dig out what happens to the Kafka container. One hack is to remove the Kafdrop from the docker compose and run only the Kafka broker so that the logs would be easy to follow.

**Comment by Bert-R** [comment]
Alternatively, run the compose file in detached mode: ``docker-compose up -d``. Now you can see the running containers through ``docker ps``. This will show you something like this:

```
CONTAINER ID   IMAGE                      COMMAND                  CREATED              STATUS          PORTS                                            NAMES
6ccbd04df405   obsidiandynamics/kafdrop   "/kafdrop.sh"            About a minute ago   Up 26 seconds   0.0.0.0:9000->9000/tcp                           kafka-kafdrop-kafdrop-1
4345843c8871   obsidiandynamics/kafka     "/bin/sh -c /opt/kaf…"   About a minute ago   Up 27 seconds   0.0.0.0:2181->2181/tcp, 0.0.0.0:9092->9092/tcp   kafka-kafdrop-kafka-1
```

To see the logs of Kafka, type ``docker logs kafka-kafdrop-kafka-1``.

**Comment by davideicardi** [comment]
Closing this for now.
